mongodump/mongodump_test.go

```diff
@@ -1865,6 +1866,8 @@ func TestFailDuringResharding(t *testing.T) {
 		md.ToolOptions.Namespace = &options.Namespace{}
 		err = md.Init()
 		So(err, ShouldBeNil)
+		// Hack to not fail validating inputs on both --db and --oplog being specified.
+		md.ToolOptions.DB = "config"
```

Comment:
Putting this after .Init() is a temporary hack I made to get around this error message: https://github.com/mongodb/mongo-tools/blob/master/mongodump/mongodump.go#L117. Without specifying config, dumping the config DB is no longer possible, so some of the downstream errors could never happen.

This is an indicator of a new limitation. We now essentially no longer support --oplog with dumping the "config" DB at all ("config" isn't dumped by default, and can't specify --db and use --oplog). I'm not totally sure that is acceptable. If that limitation is acceptable, it might be better for me to truncate this test heavily (or remove it).